lib/puppet/provider/package/xbps.rb

```diff
@@ -0,0 +1,128 @@
+require_relative "../../../puppet/provider/package"
+require "set"
+require "uri"
+
+Puppet::Type.type(:package).provide :xbps, :parent => Puppet::Provider::Package do
+  desc "Support for the Package Manager Utility (xbps) used in VoidLinux.
+
+  This provider supports the `install_options` attribute, which allows command-line flags to be passed to xbps-install.
+  These options should be specified as an array where each element is either a string or a hash."
+
+  commands :xbps_install => "/usr/bin/xbps-install"
+  commands :xbps_remove => "/usr/bin/xbps-remove"
+  commands :xbps_query => "/usr/bin/xbps-query"
+  commands :xbps_pkgdb => "/usr/bin/xbps-pkgdb"
+
+  confine :operatingsystem => [:void]
```

Comment:
```suggestion
  confine 'os.name' => :void
```

Reply:
Is this a recent change? I wasn't aware of this syntax 👀

Reply:
yeah, in puppet 8 we no longer rely on legacy facts https://github.com/puppetlabs/puppet/wiki/Puppet-8-Compatibility#legacy-facts